jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/QueuedPool.java

```diff
@@ -0,0 +1,245 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.stream.Stream;
+
+/**
+ * This implementation does not support multiplexing.
+ *
+ * @param <P> the type of the pooled objects
+ */
+public class QueuedPool<P> implements Pool<P>
```

Comment:
I don't think we need to make it private.  It is a perfectly good pool implementation that just happens not to take a multiplex int function, so it can't do multiplexing.